kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/SessionsResource.scala

```diff
@@ -88,12 +104,27 @@ private[v1] class SessionsResource extends ApiRequestContext {
   @DELETE
   @Path("{sessionHandle}")
   def closeSession(@PathParam("sessionHandle") sessionHandleStr: String): Response = {
-    val splitSessionHandle = sessionHandleStr.split("\\|")
-    val handleIdentifier = new HandleIdentifier(
-      UUID.fromString(splitSessionHandle(0)), UUID.fromString(splitSessionHandle(1)))
-    val protocolVersion = TProtocolVersion.findByValue(splitSessionHandle(2).toInt)
-    val sessionHandle = new SessionHandle(handleIdentifier, protocolVersion)
+    val sessionHandle = getSessionHandle(sessionHandleStr)
     backendService.closeSession(sessionHandle)
     Response.ok().build()
   }
+
+  def getSessionHandle(sessionHandleStr: String): SessionHandle = {
+    try {
+      val splitSessionHandle = sessionHandleStr.split("\\|")
+      val handleIdentifier = new HandleIdentifier(
+        UUID.fromString(splitSessionHandle(0)), UUID.fromString(splitSessionHandle(1)))
+      val protocolVersion = TProtocolVersion.findByValue(splitSessionHandle(2).toInt)
```

Comment:
@simon824 can you send a PR to remove TProtocolVersion for REST API. the client protocol is decided by client libraries that users use, not by users explicitly. So it's odd here to use both as a user-specified part of SessionHandle and also it is not related to the REST API at all. Also cc @yanghua

Reply:
The session handle is composed of `TProtocolVersion `. And it is the index of a session for the rest client. Here, `TProtocolVersion` is used to parse from a string presentation and recover an instance of a `SessionHandle`. Do you mean encapsulate the marshaled logic between string and `SessionHandle` object into `SessionHandle` itself, so that we can hide it from the rest component?

Reply:
For a session handle instance, TProtocolVersion is not necessary, as you can see the overridden equal/hashCode funcs. It's mainly used to decide the compatibility for kyuubi server and its thrift client, e.g. the result layout. I am not very familiar with REST API, does the `v1` also indicate the compatibility thing between rest clients and servers?

Reply:
restful API tries to interact with Kyuubi Server AFAP. It may provide a full capacity like the thrift protocol has provided (create sessions then fetch them). On the other hand, it also may provide management for Kyuubi server at any time, which means it would not create them but can fetch and interact with them. So it should support the session handles which contain  `TProtocolVersion `?